Merge configuration into configuration package

chromatic opened the pull request:
Fixes #24.

### Motivation

This removes near-duplicate code and will make it easier to replace bitcoin-specific configuration (especially data structure names and hard-coded strings) with dogecoin-specific configuration.

### Solution

A previous approach tried to merge configuration into the dogecoin-specific configuration file, but it's easier to go the other way and it requires less churn in other packages which use this configuration.

patricklodder reviewed (changes requested):
Indexer works with this.

Needs linking of the Dogecoin network parameters instead of the bitcoin ones in `LoadConfiguration()`, as was previously done within the `dogecoin/` folder.

chromatic commented:
I've moved other configuration files from `dogecoin/` into the `configuration/` directory and fixed up the references I could find. `make build-local` succeeds for me, though `make test` doesn't.
